tests/scripts/all.sh

```diff
@@ -1577,6 +1577,57 @@ component_test_full_no_cipher_with_crypto_config() {
     common_test_full_no_cipher_with_psa_crypto 1 "full no CIPHER"
 }
 
+component_test_full_no_ccm() {
+    msg "build: full no PSA_WANT_ALG_CCM"
+
+    # Full config enables:
+    # - USE_PSA_CRYPTO so that TLS code dispatches cipher/AEAD to PSA
+    # - CRYPTO_CONFIG so that PSA_WANT config symbols are evaluated
+    scripts/config.py full
+
+    # Disable PSA_WANT_ALG_CCM so that CCM is not supported in PSA. CCM_C is still
+    # enabled, but not used from TLS since USE_PSA is set.
+    # This is helpful to ensure that TLS tests below have proper dependencies.
+    #
+    # Note: also PSA_WANT_ALG_CCM_STAR_NO_TAG is enabled, but it does not cause
+    # PSA_WANT_ALG_CCM to be re-enabled.
+    scripts/config.py -f "$CRYPTO_CONFIG_H" unset PSA_WANT_ALG_CCM
+
+    make tests
+
+    msg "test: full no PSA_WANT_ALG_CCM"
+    ( cd tests; ./test_suite_ssl )
+}
+
+component_test_full_no_ccm_star_no_tag() {
```

Comment:
Future-looking note: I think this component will become redundant with the "cipher light only" component (or whatever it's called) when we do #8483 and we'll be able to remove it at that point. Let's keep it until then :)